UPGRADE.md

```diff
@@ -1,3 +1,17 @@
+4.x to 5.0
+==========
+
+* `mount` has been split into three methods:
+    * `mount` for mounting functions
+    * `mountClass` for mounting `StandaloneComponent`s
+    * `mountJsx` for mounting Preact components (functional and class)
+* Corresponding to the changes in `mount`, we had to change the mounting option types to add support for automatic parameter inference:
+    * `mojave.ClassMountOptions` has been made generic: `mojave.ClassMountOptions<T extends mojave.MountableClass>`
+    * `mojave.FunctionMountOptions` has been made generic: `mojave.FunctionMountOptions<T extends mojave.MountableFunction>`
+    * `mojave.ComponentMountOptions` has been made generic: `mojave.ComponentMountOptions<T extends ComponentClass<any> | FunctionComponent<any>>`
```

Comment:
You should not copy the changelog here (as that's what the changelog is for), but instead describe on how to upgrade (that's why this file is called "upgrade").